firebase-firestore/src/main/java/com/google/firebase/firestore/util/Util.java

```diff
@@ -298,25 +297,6 @@ public static StringBuilder repeatSequence(
     diffCollections(beforeEntries.iterator(), afterEntries.iterator(), comparator, onAdd, onRemove);
   }
 
-  /**
-   * Compares two sorted sets for equality using their natural ordering. The method computes the
-   * intersection and invokes `onAdd` for every element that is in `after` but not `before`.
-   * `onRemove` is invoked for every element in `before` but missing from `after`.
-   *
-   * <p>The method creates a copy of both `before` and `after` and runs in O(n log n), where n is
-   * the size of the two lists.
-   *
-   * @param before The elements that exist in the original set.
-   * @param after The elements to diff against the original set.
-   * @param onAdd A function to invoke for every element that is part of `after` but not `before`.
-   * @param onRemove A function to invoke for every element that is part of `before` but not
-   *     `after`.
-   */
-  public static <T extends Comparable<T>> void diffCollections(
-      SortedSet<T> before, SortedSet<T> after, Consumer<T> onAdd, Consumer<T> onRemove) {
-    diffCollections(before.iterator(), after.iterator(), before.comparator(), onAdd, onRemove);
```

Comment:
This overload is actually different from the one we now default to. This method takes sorted data - the method you are calling now reorders the data (and copies it) even when already sorted. I think you solve the problem you encountered - that `SortedSet<T>.comparator()` can return `null` - by creating a default comparator based on `T.compareTo()` if the set's comparator is null.

Reply:
Good catch, thanks. Added the code back and added a null check.